File: Dante1/Module1.06/Zadanie6.11/main.c
#include <stdio.h>
#include <math.h>

int main()
{
    int matrix[10][10];

    printf("Enter data:\n");

    int max = -__INT_MAX__;
    int max_i;
    int max_j;

    int min = __INT_MAX__;
    int min_i;
    int min_j;

    int sum = 0;

    for(int i = 0; i < 10; i++)
    {
        for(int j = 0; j < 10; j++)
        {
            int check = scanf("%i", &matrix[i][j]);
            if(check != 1)
            {
                printf("Incorrect input\n");
                return 1;
            }

            sum += matrix[i][j];
            if(matrix[i][j] > max)
            {
                max = matrix[i][j];
                max_i = i;
                max_j = j;
            }
            if(matrix[i][j] < min)
            {
                min = matrix[i][j];
                min_i = i;
                min_j = j;
            }
        }
    }

    int middle = sum / (10 * 10);

    int count1 = 0, count2 = 0, count3 = 0;
    int middle_i, middle_j;

    for (int i = 0; i < 10; i++)
    {
        for (int j = 0; j < 10; j++)
        {
            if(matrix[i][j] == max)
            {
                count1++;
            }
            if(matrix[i][j] == min)
            {
                count2++;
            }
            if(matrix[i][j] == middle)
            {
                count3++;
                middle_i = i;
                middle_j = j;
            }
        }
    }

    if(count1 != 1 || count2 != 1 || count3 != 1)
    {
        printf("Error\n");
        return 2;
    }

    float dist1 = sqrtf(powf(min_i - max_i, 2) + powf(min_j - max_j, 2));
    float dist2 = sqrtf(powf(middle_i - min_i, 2) + powf(middle_j - min_j, 2));
    float dist3 = sqrtf(powf(max_i - middle_i, 2) + powf(max_j - middle_j, 2));

    printf("%.2f %.2f %.2f\n", dist1, dist2, dist3);

    return 0;
}

File: Dante1/Module1.06/Zadanie6.11/main.rs
fn main()
{
    let mut array:[[i32; 10]; 10] = [[0; 10]; 10];
    println!("Enter: ");

    let mut max: i32 = i32::MAX;
    let mut max_i: i32 = 0;
    let mut max_j: i32 = 0;

    let mut min: i32 = i32::MIN;
    let mut min_i: i32 = 0;
    let mut min_j: i32 = 0;

    let mut sum: i32 = 0;

    let mut input: String = String::new();
    for i in 0..10
    {
        for j in 0..10
        {
            input.clear();
            std::io::stdin().read_line(&mut input).expect("Incorrect input");
            array[i][j] = input.trim().parse().expect("Incorrect parsing");

            sum += array[i][j];
            if max < array[i][j]
            {
                max = array[i][j];
                max_i = i as i32;
                max_j = j as i32;
            }
            if min > array[i][j]
            {
                min = array[i][j];
                min_i = i as i32;
                min_j = j as i32;
            }
        }
    }

    let middle: i32 = sum / 100;
    let mut counter1: i32 = 0;
    let mut counter2: i32 = 0;
    let mut counter3: i32 = 0;

    let mut middle_i: i32 = 0;
    let mut middle_j: i32 = 0;

    for i in 0..10
    {
        for j in 0..10
        {
            if array[i][j] == min
            {
                counter1 += 1;
            }
            if array[i][j] == max
            {
                counter2 += 1;
            }
            if array[i][j] == middle
            {
                counter3 += 1;

                middle_i = i as i32;
                middle_j = j as i32;
            }
        }
    }

    if counter1 == 0 || counter2 == 0 || counter3 == 0
    {
        panic!("Error!");
    }

    let dist1: f32 = f32::sqrt(f32::powi((min_i - max_i) as f32, 2) - f32::powi((min_j - max_j) as f32, 2));
    let dist2: f32 = f32::sqrt(f32::powi((middle_i - min_i) as f32, 2) - f32::powi((middle_j - min_j) as f32, 2));
    let dist3: f32 = f32::sqrt(f32::powi((max_i - middle_i) as f32, 2) - f32::powi((max_j - middle_j) as f32, 2));
    
    println!("{} {} {}", dist1, dist2, dist3);
}
